Support for Batch Requests

dadheech115 opened the pull request:
- Added classes to support Batch Requests as part of Graph Content.
- Added unit tests for these classes.

Commit:
Added classes to support Batch Requests and Unit tests for those classes.

Commit:
Merge branch 'dev' into vidadhee/BatchRequest
Resolve possible merge conflicts
# Conflicts:
#	MSGraphCoreSDK/MSGraphCoreSDK.xcodeproj/project.pbxproj

deepak2016 commented on MSGraphCoreSDK/MSGraphCoreSDKTests/GraphContent/BatchContent/MSBatchRequestContentTests.m:
```diff
@@ -0,0 +1,193 @@
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+}
+
+- (void)testInit {
```
can you give it some other name, reflecting the test case?

dadheech115 replied:
Done.

deepak2016 commented on MSGraphCoreSDK/MSGraphCoreSDKTests/GraphContent/BatchContent/MSBatchRequestContentTests.m:
```diff
@@ -0,0 +1,193 @@
+    XCTAssertNil(error);
+
+    NSMutableArray *batchStepArray = [NSMutableArray new];
+    for(int i=0;i<21;i++) {
```
please put comment why 21 is significant here

dadheech115 replied:
Done.

deepak2016 commented on MSGraphCoreSDK/MSGraphCoreSDKTests/GraphContent/BatchContent/MSBatchRequestContentTests.m:
```diff
@@ -0,0 +1,193 @@
+    error = nil;
+
+    //Test for empty request id
+    MSBatchRequestStep *empty = [[MSBatchRequestStep alloc] initWithId:@"" request:self.requestForMock andDependsOn:nil];
```
can you create different tests for each test cases?

dadheech115 replied:
Done.

Commit:
Resolved review comments-
- Broke down add batch request step test case in multiple methods
- Changed test case method name for init test and broke it in multiple cases
- Added comment for testing adding 20 batch steps test case

deepak2016 commented on MSGraphCoreSDK/MSGraphCoreSDK/GraphContent/BatchContent/MSBatchRequestContent.m:
```diff
@@ -0,0 +1,168 @@
+    self = [super init];
+    if(self)
+    {
+        maxNumberOfRequests = 20;
```
can you put a comment that 20 is a enforcement by graph

dadheech115 replied:
done.

deepak2016 commented on MSGraphCoreSDK/MSGraphCoreSDK/GraphContent/BatchContent/MSBatchRequestContent.m:
```diff
@@ -0,0 +1,168 @@
+    {
+        maxNumberOfRequests = 20;
+        //Check whether number of request steps in array are more than the limit
+        if(batchRequestStepsArray.count>maxNumberOfRequests)
```
minor: space surrounding > operator?

dadheech115 replied:
done.

deepak2016 commented on MSGraphCoreSDK/MSGraphCoreSDK/GraphContent/BatchContent/MSBatchRequestContent.m:
```diff
@@ -0,0 +1,168 @@
+- (void)removeBatchRequesStepWithId:(NSString *)requestId error:(NSError *__autoreleasing *)error
+{
+    BOOL didFindStep = NO;
+    for (long i = _batchRequestStepsArray.count-1 ; i >= 0 ; i--)
```
minor: no space before ; character?

dadheech115 replied:
done.

deepak2016 commented on MSGraphCoreSDK/MSGraphCoreSDK/GraphContent/BatchContent/MSBatchRequestContent.m:
```diff
@@ -0,0 +1,168 @@
+    for (long i = _batchRequestStepsArray.count-1 ; i >= 0 ; i--)
+    {
+        MSBatchRequestStep *requestStep = _batchRequestStepsArray[i];
+        for (long j = requestStep.arrayOfDependsOnIds.count-1 ; j >= 0 ; j--)
```
minor: no space before ; character?

dadheech115 replied:
done.

Commit:
Resolved further review comments -
- Added comment on the limit of maximum number of requests
- Added space around > character
- Removed extra space after ; character at 4 places
